src/hdmf/data_utils.py

```diff
@@ -624,7 +628,11 @@ def __len__(self):
         return len(self.data)
 
     def __bool__(self):
```

Comment:
It would be nice to add a docstring to describe what the logic behind ``__bool__`` is. Is this for checking whether DataIO holds any data?